### PR TITLE
[WIP] Fix LazyEnumerator#first without argument

### PR DESCRIPTION
**Pull request by kachick**
Reasonable direction? 🤔

### CODE REVIEW

File: lib/lazy_enumerator.gb [hunk 1]
```diff
@@ -74,7 +75,16 @@ class LazyEnumerator
 
   # Returns the first (`size`) elements.
   #
-  def first(size)
+  def first(size = NONE)
```
Comment (saveriomiroddi) [inline lib/lazy_enumerator.gb]: `size = 1` will do the trick without adding extra code :smile:

**Comment by st0012** [comment]
@kachick Agree with @saveriomiroddi. is there any reason to use `Object.new` as a default value?

File: lib/lazy_enumerator.gb (hunk 1 above)
Comment (kachick) [reply lib/lazy_enumerator.gb]: 😮 I thought `size = 1` returns `[5]` instead of `5`.

Comment (st0012) [reply lib/lazy_enumerator.gb]: I think we still need the logic you added, but can change `size == NONE` to `size == 1`

Comment (kachick) [reply lib/lazy_enumerator.gb]:
```
irb(main):001:0> [5, 6, 4].lazy.first 1
=> [5]
irb(main):002:0> [5, 6, 4].lazy.first
=> 5
irb(main):004:0> RUBY_VERSION
=> "2.4.2"
```


No argument and 1 will return another value 🤔 Sorry I might be misunderstanding your pointing outs :bow:
I'll re-consider ~

**Comment by st0012** [comment]
@kachick is it possible to make this merge by the end of this week?

**Comment by st0012** [comment]
@kachick do you plan to continue this PR?

**Comment by st0012** [comment]
@kachick I'm closing this now, you can reopen it whenever you want

**Comment by kachick** [comment]
Oh... sorry for overlooking your reply🙇‍♂️

**Comment by st0012** [comment]
no worries 😊

**Pull request closed without merging**